### ISSUE TITLE
Mistake in pushing, now can't do ANYTHING to fix it as says over LFS limit.

### ISSUE THREAD

**Issue by harlandpj** [desktop/desktop#14436]
Hi,

I made a dumb mistake on my repo "Wunderland City", a directory Assets/TerrainSampleAssets (which is 1.88 GB) on my machine was in the wrong location so got ignored by the .gitignore file, and was included in the commit pushed to the repo.

I CANNOT now undo (revert) this as I get a message about being over LFS limit and account holder should contact to buy quota.

How do I go about getting rid of the last two commits (one was a simple merge of development into main I think, as I hadn't got the message about being over the LFS quota) and the immediately previous commit, without having to go create ANOTHER repo and losing all the history?

Thanks,
Phil :-)

p.s. It would be a very useful feature to  ADD something which will check, and NOTIFY you BEFORE you push something that you will go over the Quota allocated, ensuring people don't make silly mistakes!

**Comment by tidy-dev** [comment]
👋  Hi, 
There is a git command you can use in the terminal, but if you this incorrectly you could lose those changes, thus, I would recommend making a copy of your repo. 

You can reset your last commit via the terminal and keep your changes with the soft flag.
```git reset --soft HEAD~1```. (either repeat or change to HEAD~2 to do a second commit)

After you have resolved your issue (removed the files you didn't want to push), you will likely need to go to the terminal and run
``` git push --force ```
to update your remote branch.


If you are unfamiliar with the terminal, a round about way you might be able to do this in GitHub Desktop.
1. Go to history find a commit in that the branch that is in a good state. (before commit of the large files)
2. Right click that commit and select "Create a branch from this  commit". (make a new branch)
3. Go to back to your affected branch and select the too large commits.
4. Drag and drop them on your new branch in the branch menu Or right click them and select cherry-pick and then select your new branch.

Now your affected commits will be local to your new branch and you should be able to undo them.

**Comment by harlandpj** [comment]
Hi,

This reply doesn't help!

Here's what I tried.

1. Started with an EMPTY directory on my PC.
2. Opened Github desktop which says can't locate repo (as expected) on my pc, and cloned it again.
3. Clone then FAILS with error "Clone succeeded, but checkout failed" due to Git repository over its data quota.

Yes I know it is, I'm trying to reduce the size of it!!!!

So... I CAN'T EVEN CLONE (even when I start with an empty directory on my PC) the repo, and when I try to switch from main to the development branch with the offending commits on, it WON'T stash changes of the stuff (which I just don't care about anyway, and won't let me overwrite them), that I had to clone to even open the repo, as my repo is.... OVER the quota (as it keeps telling me EVERY single time I try) using Github Desktop, so how can I do the changes you suggest?


Thanks!
Thanks
Phil
